### PR TITLE
[Hazmat Shipping] Handle hazmat category selection

### LINKED ISSUE #10584: [Hazmat Shipping] Introduce Hazmat category selection into Shipping Label creation
Once the Hazmat section UI is available through #10551, we need to properly allow the selection of the hazmat category through a selection list, following the same possible options offered by the web implementation.

image

### PR DESCRIPTION
**Pull request by ThomazFB**
Closes #10584

Why
==========
During the Hazmat shipping declaration, it's mandatory to describe in which category the hazardous material falls in. So whenever the Hazmat section is expanded, we should allow the merchant to select the correct category.

This PR introduces the Hazmat category selection UI interaction, but still with no integration with models and network operations. So when we leave the package details view, the selection is lost.

How
==========
Introduces the hazmat category selection flow that's not connected to no models or network operation yet by defining a Button layout containing a flexible area to allow displaying which hazmat category is currently selected, or asking the merchant to select one.

Screen Capture
==========
https://github.com/woocommerce/woocommerce-ios/assets/5920403/a7908a53-ee29-4d76-a1fd-c5d055160f45

How to Test
==========
1. Open the order details of an order eligible for Shipping label creation
2. Start the shipping label creation flow until you reach the Package details section
3. Inside the package details view, make sure the Contains hazardous materials toggle is visible and interactable
4. Make sure the toggle expands and collapses a basic view that will serve as the main control options for the HAZMAT declaration
5. Hit the `Hazardous material category` section, make sure it works as expected for a selection field, and allow selecting or changing the category any time.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/ShippingLabelHazmatCategory.swift [hunk 1]
```diff
@@ -0,0 +1,161 @@
+import Foundation
+
+enum ShippingLabelHazmatCategory: String, CaseIterable {
+    case none
+    case airEligibleEthanol
+    case class1
+    case class3
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/ShippingLabelHazmatCategory.swift:7]: just wondering, there is no class 2? 😆

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/ShippingLabelSinglePackageViewModel.swift [hunk 2]
```diff
@@ -124,6 +128,10 @@ final class ShippingLabelSinglePackageViewModel: ObservableObject, Identifiable
     ///
     @Published private var isPackageWeightEdited: Bool = false
 
+    /// The group of selectable Hazardous materials that can be selected
+    ///
+    let selectableHazmatCategories = ShippingLabelHazmatCategory.allCases.filter { $0 != .none }
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/ShippingLabelSinglePackageViewModel.swift:133]: nit: might be worth having a test case to ensure the categories don't include `none`

**Review by jaclync — APPROVED** [review]
Looks good! 🚢 

Non-blocking: the alignment looks a bit funny

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/ShippingLabelHazmatCategory.swift (hunk 1 above)
Comment (ThomazFB) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/ShippingLabelHazmatCategory.swift:7]: Weird, right? 😆 I also had to double check the [easypost API docs](https://www.easypost.com/docs/api) to make sure there's no class 2.
image

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/ShippingLabelSinglePackageViewModel.swift (hunk 2 above)
Comment (ThomazFB) [reply WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/ShippingLabelSinglePackageViewModel.swift:133]: Indeed! Added the unit test for this scenario in https://github.com/woocommerce/woocommerce-ios/pull/10585/commits/a3044963904b64b5b4ae0953d76b56e494800852 and also added another one to make sure the ViewModel always start with the `.none` selected in https://github.com/woocommerce/woocommerce-ios/pull/10585/commits/820e5df1084208028abe3234660b7bbf531a52a8, covering the entire category scenario.

**Comment by ThomazFB** [comment]
@jaclync Thank you for the review! I addressed all your suggestions and the PR is ready for a second round! I just want to check with you about the Selection list text alignment:

I also found odd the alignment like that, and previously verified that just adding the `multilineTextAlignment(.leading)` inside the `SelectableItemRow` fixes the problem. What stopped me was the fear of changing something like this in a component that is used in multiple points of the app, and I don't know if the `multilineTextAlignment` have any caveats I'm not aware about. Do you think it's safe to proceed? I submitted the fix in https://github.com/woocommerce/woocommerce-ios/pull/10585/commits/5c7aea0e892069c4ee55cc6f16dccfda2e1af33d.

**Comment by jaclync** [comment]
> I also found odd the alignment like that, and previously verified that just adding the multilineTextAlignment(.leading) inside the SelectableItemRow fixes the problem. What stopped me was the fear of changing something like this in a component that is used in multiple points of the app, and I don't know if the multilineTextAlignment have any caveats I'm not aware about. Do you think it's safe to proceed? I submitted the fix in https://github.com/woocommerce/woocommerce-ios/commit/5c7aea0e892069c4ee55cc6f16dccfda2e1af33d.

Yea, this change affects other screens and it's worth testing these affected views. I tested the SwiftUI views that include `SelectableItemRow` (`ShippingLabelPackageList`, `ShippingLabelServicePackageList`, `ShippingLabelPaymentMethods`) and the multiline text looks good with the change in these views 👍 Feel free to merge it!

**Pull request merged**